### ISSUE TITLE
Support out of source tree builds

### ISSUE THREAD

**Issue by nickbroon** [gcovr/gcovr#61]
This does not work for me:

```
cd /tmp/builds/project1;
/bin/python /home/user1/gcovr/scripts/gcovr -v -x -r /home/user1/project1 --object directory=/tmp/builds/project1 -o /home/user1/coverage.xml
```

I get the following output:

```
Scanning directory /home/user1/project1 for gcda/gcno files...
Found 0 files (and will process 0)
Gathered coveraged data for 0 files
```

I expected `gcovr` to use the path given by the `--object-directory` option to look for gcda/gcna files, while using the `-r` option to specify where the source files are, but it appears just look in the path given by `-r`.

**Comment by lemonkey** [comment]
I too can confirm this is happening with the latest (v3.2).  If I copy my .gcno files into the same root folder as the source files defined by the -r argument, it generates the report just fine.  But if I define the object-directory path to be somewhere else, the report processes 0 files.

**Comment by briangreenery** [comment]
I'm not sure it'll work for you, but this is the solution I found: https://github.com/gcovr/gcovr/issues/31#issuecomment-48809388

Basically, inside the build directory run:

    gcovr --root=. --filter=.*

Maybe https://github.com/gcovr/gcovr/pull/72 fixes this?

**Comment by wangshan** [comment]
it's not very clear what --object-directory does, but it may work if you remove it, just specify the directory where the .gcda, .o and .gcno files are, like this: /bin/python /home/user1/gcovr/scripts/gcovr -v -x -r /home/user1/project1 /tmp/builds/project1 -o /home/user1/coverage.xml

**Comment by nickbroon** [comment]
I wonder if #108 relates to this?

**Comment by latk** [comment]
This issue has not seen any progress in a while, so I am closing it for now. If the problem persists, please add further details so it can be re-opened. If someone else is experiencing a similar issue, please create a new issue.

I suppose this was fixed in 3.3 or 3.4. The `--object-directory` is a bit special and overrides normal path detection. In most cases specifying the build directory as an (unnamed) argument is sufficient.
